### PR TITLE
Tip 475 : change the use of the structured format to the standard format for media

### PR DESCRIPTION
**Pull request by ahocquard**
**Description (for Contributor and Core Developer)**

The structure of the media data has changed with the standard format.

Before, the FileNormalizer returned the key (=file path), the original filename, and the hash : 
```
[
    filePath:    
    originalFileName :
    hash :
]
```

From now on, it returns only the key.

The goal of this PR is to adapt this new structure in the backend & frontend.

To avoid major impacts on the frontend side, 1 converter are charged to transform the standard format into the structured format (with originalFilename), and vice-versa.
 
**Definition Of Done (for Core Developer only)**

| Q | A |
| --- | --- |
| Added Specs | y |
| Added Behats | y |
| Changelog updated |  y|
| Review and 2 GTM |  |
| Micro Demo to the PO (Story only) |  |
| Migration script |  |
|  Tech Doc |  |

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/VariantGroupController.php [hunk 1]
```diff
@@ -57,16 +58,21 @@ class VariantGroupController
...
-     * @param EntityRepository            $repository
-     * @param NormalizerInterface         $normalizer
-     * @param ObjectUpdaterInterface      $updater
-     * @param SaverInterface              $saver
-     * @param UserContext                 $userContext
-     * @param AttributeConverterInterface $attributeConverter
-     * @param ValidatorInterface          $validator
-     * @param NormalizerInterface         $violationNormalizer
-     * @param CollectionFilterInterface   $variantGroupDataFilter
+     * @param EntityRepository|GroupRepositoryInterface $repository
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/VariantGroupController.php]: you can remove EntityRepository

File: src/Pim/Component/Catalog/Comparator/Attribute/FileComparator.php [hunk 2]
```diff
@@ -43,24 +43,31 @@ public function supports($type)
...
-        // compare a local file and a stored file (can happen during an import for instance)
-        if (isset($data['data']['filePath']) &&
-            isset($originals['data']['filePath']) &&
-            is_file($data['data']['filePath'])
-        ) {
-            $originalFile = $this->repository->findOneByIdentifier($originals['data']['filePath']);
-            if (null !== $originalFile &&
-                $originalFile->getHash() === $this->getHashFile($data['data']['filePath'])) {
+        // compare a local or stored file with a stored file (can happen during an import for instance)
+        if (isset($data['data']) && isset($originals['data'])) {
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Comparator/Attribute/FileComparator.php]: useless condition as you already check `if (!isset($data['data']) && !isset($originals['data']))`

File: src/Pim/Component/Catalog/Comparator/Attribute/FileComparator.php [hunk 3]
```diff
@@ -43,24 +43,31 @@ public function supports($type)
...
+            $originalFile = $this->repository->findOneByIdentifier($originals['data']);
+
+            if (is_file($data['data']) &&
+                null !== $originalFile &&
+                $originalFile->getHash() === $this->getHashFile($data['data'])
+            ) {
                 return null;
+            } else {
```
Comment (skeleton) [inline src/Pim/Component/Catalog/Comparator/Attribute/FileComparator.php]: else is useless as there is an early return in your previous condition

File: src/Pim/Component/Catalog/spec/Comparator/Attribute/FileComparatorSpec.php [hunk 4]
```diff
@@ -22,72 +22,85 @@ function it_is_initializable()
...
-        $this->compare(
             ['data' => null],
             ['data' => null]
-        )->shouldReturn(null);
-
-        $this->compare(
-            ['data' => ['filePath' => null]],
-            ['data' => null]
-        )->shouldReturn(null);
+        )->shouldReturn(null) ;
```
Comment (skeleton) [inline src/Pim/Component/Catalog/spec/Comparator/Attribute/FileComparatorSpec.php]: extra space :trollface:

File: src/Pim/Component/Enrich/Converter/ConverterInterface.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace Pim\Component\Enrich\Converter;
+
+/**
```
Comment (skeleton) [inline src/Pim/Component/Enrich/Converter/ConverterInterface.php]: sorry, I forget to add a real description :( Can you add it please ? Explain the standard format has to be convert to be understanding by the front

**Comment by SamirBoulil** [comment]
GTM After fix

File: src/Pim/Component/Enrich/Converter/ConverterInterface.php (hunk 5 above)
Comment (ahocquard) [inline src/Pim/Component/Enrich/Converter/ConverterInterface.php]: done

**Pull request merged**